### PR TITLE
[SPARK-12456][SQL] Add ExpressionDescription to misc functions

### PR DESCRIPTION
**Pull request by xguo27**
First try, not sure how much information we need to provide in the usage part.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/InputFileName.scala [hunk 1]
```diff
@@ -26,6 +26,9 @@ import org.apache.spark.unsafe.types.UTF8String
 /**
  * Expression that returns the name of the current file being read in using [[SqlNewHadoopRDD]]
  */
+@ExpressionDescription(
+  usage = "_FUNC_() - Returns the name of the current file being read in using [SqlNewHadoopRDD]",
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/InputFileName.scala]: "in using [SqlNewHadoopRDD]" -> "if available"

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/MonotonicallyIncreasingID.scala [hunk 2]
```diff
@@ -32,6 +32,9 @@ import org.apache.spark.sql.types.{LongType, DataType}
  *
  * Since this expression is stateful, it cannot be a case object.
  */
+@ExpressionDescription(
+  usage = "_FUNC_() - Returns monotonically increasing 64-bit integers",
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/MonotonicallyIncreasingID.scala]: You should probably include the entire paragraph up there.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala [hunk 3]
```diff
@@ -53,6 +56,11 @@ case class Md5(child: Expression) extends UnaryExpression with ImplicitCastInput
  * asking for an unsupported SHA function, the return value is NULL. If either argument is NULL or
  * the hash length is not one of the permitted values, the return value is NULL.
  */
+@ExpressionDescription(
+  usage = "_FUNC_(input) - Returns a checksum of SHA-2 family as a hex string of the input. " +
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala]: this accepts two args?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala [hunk 4]
```diff
@@ -30,6 +30,9 @@ import org.apache.spark.unsafe.types.UTF8String
  * A function that calculates an MD5 128-bit checksum and returns it as a hex string
  * For input of type [[BinaryType]]
  */
+@ExpressionDescription(
+  usage = "_FUNC_(input) - Returns an MD5 128-bit checksum as a hex string of the input",
+  extended = "> SELECT _FUNC_('SparkSql');\n '49df15ea9f84f6fcf94f2823bf3f861e'")
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala]: I'd prefer to just use "Spark".

**Comment by xguo27** [comment]
@rxin Thank you very much for go through the changeset, Reynold! I have updated it per your suggestions.

**Comment by rxin** [comment]
What's your JIRA id?

**Comment by xguo27** [comment]
@rxin Great, thanks Reynold! My JIRA id is xguo27.

**Pull request closed without merging**